asf_lifetime_cost_model/pipeline/cost_computations.py

```diff
@@ -0,0 +1,236 @@
+
+    Returns:
+        float: The total maintenance costs over the specified number of years.
+    """
+    return annual_maintenance_costs * life_span
+
+
+def creates_cost_reduction_data(
+    annual_cost_reduction: float,
+    reference_year: int = config.get("cost_data_reference_year"),
+    max_year: int = config.get("cost_year_max"),
+) -> dict:
+    """Creates cost reduction data for a heating system over the years by applying an annual cost reduction rate.
+
+    These values can be used to adjust the installation costs of a heating system based on the cost of
+    installation is a specific reference year.
+
+    Args:
+        annual_cost_reduction (float): The annual cost reduction rate. Should be a value between 0 and 1.
+        reference_year (int, optional): The year from which the reduction starts.
+            Defaults to config.get("cost_data_reference_year").
+        max_year (int, optional): The maximum year for which the reduction is calculated.
+            Defaults to config.get("cost_year_max").
+
+    Returns:
+        dict: A dictionary with years as keys and the reduction factor as values.
+    """
+    reduction_per_year = {reference_year: 1}  # no reduction in the reference year
+
+    for year in range(reference_year + 1, max_year + 1):
+        reduction_per_year[year] = (1 - annual_cost_reduction) * reduction_per_year[year - 1]
+
+    return reduction_per_year
+
+
+def get_installation_cost(heating_system: str, archetype: str, decile: int = None) -> float:
+    """Gets the cost of a heating system for a specific archetype (and decile where applicable).
+
+    Args:
+        heating_system (str): heating system.
+            Takes "ashp" (for air source heat pump) or "boiler" (for gas boiler).
+        archetype (str): Propery archetype
+        decile (int, optional): cost decile, only used for air source heat pumps.
+            Takes multiples of 10 between 10 and 90, inclusive.
+
+    Raises:
+        ValueError: If the heating system inputed is not supported
+                    or the decile is not a multiple of 10 between 10 and 90.
+
+    Returns:
+        float: The cost of the heating system for the specified archetype (and decile where applicable).
+    """
+    if decile is not None and (decile < 10 or decile > 90 or decile % 10 != 0):
+        raise ValueError("Decile must be a multiple of 10 between 10 and 90, inclusive.")
+
+    if heating_system == "ashp":
+        costs_data = data_getters.get_ashp_installation_costs()
+        cost_value = costs_data[costs_data["archetype_label"] == archetype][f"cost_percentile_{decile}"].iloc[0]
+    elif heating_system == "boiler":
+        costs_data = data_getters.get_gas_boiler_installation_costs()
+        cost_value = costs_data[(costs_data["archetype_label"] == archetype)]["cost"].iloc[0]
+    else:
+        raise ValueError(f"Unsupported heating system: {heating_system}")
```

Comment:
```suggestion
        raise ValueError(f"Unsupported heating system: {heating_system}. Supported heating systems are {list(config.get("life_span_default").keys())}"
```

Suggest to add a line on which heating system strings are valid. I'm assuming that any new heating system we introduce later will also have its own lifespan that will be added to the `life_span_default` definition in config.